### DISCUSSION TITLE
Load environment variables from dotenv with v7 and Symfony 6

### DISCUSSION THREAD

**Discussion by ToshY** [deployphp/deployer#2866] [Help needed]
It was suggested to use `deployer/dist` in #2860, but that leaves me with the problem that I don't know how to load enviroment variables from my `.env` file properly anymore.

Previously in Symfony 5, using `deployer/deployer:v7.0.0-rc.3`, a possible solution was provided [here](https://github.com/deployphp/deployer/issues/1446#issuecomment-364914342), which was suggesting using [symfony/dotenv](https://github.com/symfony/dotenv) to load the environment variables.

```php
<?php

namespace Deployer;

use Symfony\Component\Dotenv\Dotenv;

require 'recipe/common.php';
require 'contrib/discord.php';

$dotenv = new Dotenv();
$dotenv->loadEnv(__DIR__ . '/.env');

set('application', $_ENV['APP_NAME']);
set('discord_channel', $_ENV['DISCORD_DEPLOY_CHANNEL']);
set('discord_token', $_ENV['DISCORD_DEPLOY_TOKEN']);
...
```

This worked, and while I'm not sure if it was the best solution for the problem at hand, it got the job done. 

Now I've tried the suggestion using `deployer/dist:v7.0.0-rc.3` (with Symfony 6), and without knowing/realizing about the possible differences between `deployer/deployer` and `deployer/dist`, I just tried to see if it worked, resulting in the following error message when trying to display help:

```shell
Class "Symfony\Component\Dotenv\Dotenv" not found
```

So now I'm just stuck on what to do next. What is a proper way to load (all) environment variables so I can use them throughout my `deploy.php`?

**Comment by antonmedv** [comment]
Include your autoload.php or use https://deployer.org/docs/7.x/recipe/common#dotenv

**Reply by ToshY** [reply]
@antonmedv Tried adding `require_once 'vendor/autoload.php';` to my `deploy.php` which didn't make any difference.

And the link you're referrencing states the following in the docs:

> Path to .env file which will be used as environment variables for each command per run().

Which is not the point of what I'm trying to achieve here, because I want to be able to use the environment variables throughout the deploy script, so I can simply `set` the application name from the variable available in the `.env` file.

Could you please elaborate on it further or maybe supply a sample of a working deploy script for Symfony 6 using `deployer/dist:v7.0.0-rc.3`?

**Reply by antonmedv** [reply]
Maybe include it right into Deployer?

**Reply by ToshY** [reply]
Sorry but could you be a bit more specific on how to do this exactly?

**Comment by youshef** [comment]
For the ones looking for an answer, you will need to manually require the autoloader using `require_once __DIR__ . '/vendor/autoload.php';`
